### PR TITLE
fix(tracks): Do not signal muted audio tracks.

### PR DESCRIPTION
**Pull request by jallamsetty1**
Do not add the muted audio tracks to peerconnection until the user unmutes the first time. This applies to startSilent, startWithAudioMuted and startAudioMuted/startVideoMuted config.js settings.

### CODE REVIEW

File: conference.js [hunk 1]
```diff
@@ -2163,8 +2167,26 @@ export default {
...
+            for (const track of localTracks) {
+                if (audioMuted && track.jitsiTrack?.getType() === MEDIA_TYPE.AUDIO) {
+                    promises.push(this.useAudioStream(null));
+                }
+                if (videoMuted && track.jitsiTrack?.getType() === MEDIA_TYPE.VIDEO) {
+                    promises.push(this.useVideoStream(null));
+                }
+            }
+
+            Promise.all(promises)
```
Comment (saghul) [inline conference.js:2188]: This will resolve if the first one fails. Maybe we want `Promise.allSettled` instead?

Comment (jallamsetty1) [reply conference.js:2188]: Yep, makes sense. Will change it to use `Promise.allSettled`.

**Comment by jallamsetty1** [comment]
The StartMutedTest test is broken because it checks for p2 to send/receive media when it is supposed to join audio/video muted. It has been passing until now since the functionality was broken in master, i.e, if a user selects` everyone starts muted`, the users joining the call after that do not join muted, instead they start sending media which is why the test was passing.

**Comment by jallamsetty1** [comment]
Jenkins please test this please.

**Review by damencho — APPROVED** [review]

**Pull request merged**